lib/iris/coords.py

```diff
@@ -1628,6 +1628,195 @@ def __hash__(self):
+            if data.shape != self.shape:
+                raise ValueError("New data shape must match existing data "
+                                 "shape.")
+
+        self._data = data
+
+    @property
+    def shape(self):
+        """The fundamental shape of the Cell Measure, expressed as a tuple."""
+        # Access the underlying _points attribute to avoid triggering
+        # a deferred load unnecessarily.
+        return self._data.shape
+
+    @property
+    def ndim(self):
+        """
+        Return the number of dimensions of the cell measure.
+
+        """
+        return len(self.shape)
+
+    @measure.setter
+    def measure(self, measure):
+        if measure not in ['area', 'volume']:
+            raise ValueError('measure must be area or volume, '
+                             'not {}'.format(measure))
+        self._measure = measure
+
+    def __getitem__(self, key):
+        """
+        Returns a new Cell_Measure whose values are obtained by
+        conventional array indexing.
+
+        """
+        # Turn the key(s) into a full slice spec - i.e. one entry for
+        # each dimension of the cell_measure.
+        full_slice = iris.util._build_full_slice_given_keys(key, self.ndim)
+
+        # If it's a "null" indexing operation (e.g. cell_measure[:, :]) then
+        # we can preserve deferred loading by avoiding promoting _data
+        # and _bounds to full ndarray instances.
+        def is_full_slice(s):
+            return isinstance(s, slice) and s == slice(None, None)
+        if all(is_full_slice(s) for s in full_slice):
+            data = self._data
+        else:
+            data = self._data
+            if isinstance(data, iris.aux_factory._LazyArray):
+                # This triggers the LazyArray to compute its values
+                # (if it hasn't already), which will also trigger any
+                # deferred loading of its dependencies.
+                data = data.view()
+
+            # Make indexing on the cube column based by using the
+            # column_slices_generator (potentially requires slicing the
+            # data multiple times).
+            _, slice_gen = iris.util.column_slices_generator(full_slice,
+                                                             self.ndim)
+            for keys in slice_gen:
+                if data is not None:
+                    data = data[keys]
+                    if data.shape and min(data.shape) == 0:
+                        raise IndexError('Cannot index with zero length '
+                                         'slice.')
+
+        new_cell_measure = self.copy(data=data)
+        return new_cell_measure
+
+    def copy(self, data=None):
+        """
+        Returns a copy of this cell_measure.
+
+        Kwargs:
+
+        * data: A data array for the new cell_measure.
+                  This may be a different shape to the data of the
+                  cell_measure being copied.
```

Comment:
@marqh Indent trauma ... and is it `cell measure`, `cell_measure` or `CellMeasure`